### ISSUE TITLE
Port publishing doesn't work with 127.0.0.1 using docker compose

### ISSUE THREAD

**Issue by dmmartins** [docker/for-mac#5809]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: C60768A3-AF96-4423-8D9B-E976C4A64C65/20210712144856

### Expected behavior

Port publishing should work on 127.0.0.1 using docker compose

### Actual behavior

Port publishing doesn't work on 127.0.0.1 using docker compose

### Information

  - macOS Version: 10.15.7
  - Intel chip or Apple chip: Intel Core i7
  - Docker Desktop Version: Version 3.5.1 (3.5.1.7)

### Steps to reproduce the behavior

```
# Start a docker compose service publishing a port
$ cat docker-compose.yml 
version: '3.7'

services:
  http:
    image: python:3.9
    command: python3 -m http.server 3000
    ports:
      - 3000:3000
 
$ docker compose up -d
[+] Running 1/1
 ⠿ Container docker-port_http_1  Started                                    3.2s

# Connecting to that port using localhost works
$ curl -v --head http://localhost:3000
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 3000 (#0)
> HEAD / HTTP/1.1
> Host: localhost:3000
> User-Agent: curl/7.64.1
> Accept: */*
> 
* HTTP 1.0, assume close after body
< HTTP/1.0 200 OK
HTTP/1.0 200 OK
< Server: SimpleHTTP/0.6 Python/3.9.6
Server: SimpleHTTP/0.6 Python/3.9.6
< Date: Wed, 30 Jun 2021 23:20:48 GMT
Date: Wed, 30 Jun 2021 23:20:48 GMT
< Content-type: text/html; charset=utf-8
Content-type: text/html; charset=utf-8
< Content-Length: 987
Content-Length: 987

< 
* Closing connection 0


# Connecting to that port using 127.0.0.1 doesn't work
$ curl -v --head http://127.0.0.1:3000
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 3000 (#0)
> HEAD / HTTP/1.1
> Host: 127.0.0.1:3000
> User-Agent: curl/7.64.1
> Accept: */*
> 
* Recv failure: Connection reset by peer
* Closing connection 0
curl: (56) Recv failure: Connection reset by peer
```

**Comment by jmarbutt** [comment]
I am also having this issue and I am on an Apple M1 and the same version of docker

**Comment by dmmartins** [comment]
this was fixed on Version 3.5.2 (3.5.2.18)
